service/telemetry.go

```diff
@@ -118,18 +98,71 @@ func (tel *telemetryInitializer) init(res *resource.Resource, settings component
 		return err
 	}
 
-	return tel.initPrometheus(res, settings.Logger, cfg.Metrics.Address, cfg.Metrics.Level, asyncErrorChannel)
+	if !tel.useOtel {
+		// when still using opencensus, return here as there's no need to configure readers below
+		_, err := tel.initPrometheus(res, settings.Logger, cfg.Metrics.Address, cfg.Metrics.Level, asyncErrorChannel)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(cfg.Metrics.Address) > 0 {
+		settings.Logger.Warn("service.telemetry.metrics.address is being deprecated in favor of service.telemetry.metrics.metric_readers")
```

Comment:
Do we need to somehow feature gate this behavior where the user has enabled the UseOTel feature gate but hasn't started using the Reader config yet?  Sounds like we eventually we want users to configure the Readers and won't support `address`, but that would be a breaking change.

Reply:
Yup, I'm planning on putting this all behind this feature gate: https://github.com/open-telemetry/opentelemetry-collector/pull/7678